### PR TITLE
Wig adjustments

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes the wig use hair code for rendering its looks, as well as allowing wigs to have gradients.
Allows for any hat that doesn't hide hair to have a wig attached. This also tremendously reduces the time it takes to strip the hat off.

## Why It's Good For The Game
Allows for more player self-expression and customization, as well as shenanigans relating to easily strippable hats.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/25d7b229-9c49-440d-9eb0-c5d728b7249a

</details>

## Changelog
:cl:
add: Added the ability to attach and detach wigs from hats that don't hide hair
add: Added the ability for wigs to choose hair gradients and gradient colors
balance: Hats with attached wigs, and wigs themselves, only take one second to be stripped off
code: Made the wigs force the handle_hair species proc to use the hair values of the wig
/:cl:

### CODE REVIEW

**Comment by WigLiz** [comment]
As a wig lizard, I approve the creation of this PR. o7

File: code/modules/clothing/head/_head.dm [hunk 1]
```diff
@@ -8,13 +8,79 @@
...
+		if (attached_wig.resistance_flags & INDESTRUCTIBLE)
+			attached_wig.forceMove(get_turf(src))
+		else
+			QDEL_NULL(attached_wig)
+	..()
+
+/obj/item/clothing/head/examine(mob/user)
+	. = ..()
+	if(attached_wig)
+		. += "<span class='notice'>There's \a [attached_wig.name] attached, which can be removed through the context menu.</span>"
```
Comment (EvilDragonfiend) [inline code/modules/clothing/head/_head.dm:82]: I have a concern that this examining message says this kind of OOC thing - some people would feel immersion break from this message.  What if we use href to remove a wig? or there might need to be another way to remove...

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
wig should have this kind of code
```DM
/obj/item/wig/Destroy()
  if(istype(loc, /obj/item/hat))
    var/obj/item/hat/wigged_hat = loc
    wigged_hat.attached_wig = null
```

This is because a wig can be removed by acid
(as far as I am right, wig item's loc becomes hat when it's attached. if not, it should have a new variable in the wig for which hat it is attached to.)

File: code/modules/clothing/head/_head.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/modules/clothing/head/_head.dm:82]: It's the case for a lot of items
image

**Review by Rukofamicom — APPROVED** [review]
As much as I dislike wig lizards, this seems like an improvement to me.

**Review by EvilDragonfiend — APPROVED** [review]
I don't see any additional possible glitch now.

**Pull request merged**